src/analysis/retail/hunter/marksmanship/modules/talents/SurgingShots.tsx

```diff
@@ -49,8 +49,14 @@ class SurgingShots extends Analyzer {
     this.damage += calculateEffectiveDamage(event, SURGING_SHOTS_DAMAGE_INCREASE);
   }
 
-  onAimedShotCast() {
+  onAimedShotCast(event: CastEvent) {
     this.aimedShotCasts += 1;
+    if (
+      this.spellUsable.isOnCooldown(SPELLS.RAPID_FIRE.id) &&
+      this.spellUsable.chargesAvailable(SPELLS.RAPID_FIRE.id) === 0
+    ) {
+      this.spellUsable.endCooldown(SPELLS.RAPID_FIRE.id, event.timestamp);
+    }
```

Comment:
My reading of the Surging Shots talent is that Aimed Shot only has a small chance to reset Rapid Fire, but this implementation always resets Rapid Fire on Aimed Shot usage. I'm sure this reduced the number of errors in the debug, but it is adding a new non-debug-error problem: we'll frequently show Rapid Fire as not on cooldown when it actually is.

My guess is there is no event in logs that indicates when a proc has occured, which unfortunately means there is no foolproof way for WoWA to see every proc. The best you can do is look for "too early" Rapid Fire usages and infer that a proc occured.

Reply:
I mostly followed this example [here](https://github.com/WoWAnalyzer/WoWAnalyzer/blob/the-war-within/src/analysis/retail/hunter/beastmastery/modules/core/SpellUsable.tsx#L77). Where it looks like it takes an optimistic approach to reset (same low percentage scenario for reset). 

You're correct, there rly is no sign that surging shots triggers. I'll remove this though, didnt think about the wider impact of showing rapid fire off cd when it could be.